### PR TITLE
[rust] remove encoders

### PR DESCRIPTION
**Pull request by jhowarth**
This removes the encoders from loqui. Loqui now just speaks in `Vec<u8>`. Loqui will negotiate the encoding but will not do any encoding itself. That is up to the user of loqui. This was the original intention as specified in the spec (and also happens to make it easier to use from rust).

This also changes the `Client::connect` function to be `Client::start_connect` and then it will return immediately. This fixes a race condition that could occur for users of the `Client` class. Users can call `client.await_ready()` to wait for the client to be ready.

### CODE REVIEW

File: rust/loqui_client/src/client.rs [hunk 1]
```diff
@@ -3,56 +3,127 @@ use crate::waiter::ResponseWaiter;
...
-pub struct Client<F: EncoderFactory> {
-    connection: Connection<ConnectionHandler<F>>,
+pub struct Client {
+    connection: Connection<ConnectionHandler>,
     request_timeout: Duration,
+    handshake_deadline: Instant,
+    ready: Arc<AtomicBool>,
+    ready_waiter_tx: Sender<oneshot::Sender<()>>,
+    // TODO: might be able to set this unsafe
+    encoding: Arc<RwLock<Option<&'static str>>>,
```
Comment (jhowarth) [inline rust/loqui_client/src/client.rs]: What do you folks think of doing an `unsafe` set of this? It's guarded by `self.encoding()`. For what it's worth, it doesn't seem to actually change the benchmarks.

Comment (rstone) [reply rust/loqui_client/src/client.rs]: I wouldn't bother introducing an `unsafe` if it doesn't meaningfully improve benchmarks.

File: rust/loqui_client/src/lib.rs [hunk 2]
```diff
@@ -5,11 +5,11 @@ extern crate log;
 
 mod client;
 mod config;
-mod connection_handler;
+// XXX: This is required to be public due to a bug in the rust compiler.
```
Comment (rstone) [inline rust/loqui_client/src/lib.rs]: wat. Is this documented somewhere? If so maybe include a link?

Comment (jhowarth) [reply rust/loqui_client/src/lib.rs]: 😬yeah, I added a link to the ticket right under.

**Pull request merged**